app/mailers/host_report_mailer.rb

```diff
@@ -0,0 +1,52 @@
+require 'uri'
+
+class HostReportMailer < ApplicationMailer
+  helper ApplicationHelper
+
+  # sends out a summary email of hosts and their metrics (e.g. how many changes failures etc).
+  def summary(options = {})
+    raise ::Foreman::Exception, N_("Must specify a valid user with email enabled") unless (user = User.find(options[:user]))
+    hosts = User.as user do
+      Host::Managed.authorized_as(user, :view_hosts, Host).order(:last_report).limit(100)
```

Comment:
You need to return the first 100 hosts sorted by last_report in descending order, not ascending. This would give you the most uninteresting hosts, not the ones which has recent reports.

Also, hosts must be filtered here by time which is an argument of the rake task. Add a `where` statement on `last_report > ?` against the time from `options[:time]`.